201-vm-vsts-agent/azuredeploy.json

```diff
@@ -1,317 +1,252 @@
+            },
+            "defaultValue": "https://raw.githubusercontent.com/mrptsai/201-vm-vsts-agent/master/scripts/Install-DevOpsAgent.ps1"
         },
-        {
-          "name": "AzureAD",
-          "version": "2.0.1.3"
+        "vm_admin_user": {
+            "type": "string",
+            "metadata": {
+                "description": "The name of the Administrator Account to be created"
+            }
         },
-        {
-          "name": "Bitbucket.v2",
-          "version": "1.1.2"
+        "vm_admin_password": {
+            "type": "securestring",
+            "metadata": {
+                "description": "The password for the Admin Account. Must be at least 12 characters long"
+            },
+            "minLength": 12
         },
-        {
-          "name": "GetPassword",
-          "version": "1.0.0.0"
+        "vm_size": {
+            "type": "string",
+            "metadata": {
+                "description": "Specifiy the size of VM required for the VM(s)"
+            },
+            "defaultValue": "Standard_DS1_v2"
         },
-        {
-          "name": "posh-git",
-          "version": "0.7.1"
-        }
-      ]
-    },
-    "publicIPDnsName": {
-      "type": "string",
-      "metadata": {
-        "description": "Enter a DNS name to resolve to the Public IP Address"
-      }
-    },
-    "vmAdminUser": {
-      "type": "string",
-      "metadata": {
-        "description": "The name of the Administrator Account to be created"
-      }
-    },
-    "vmAdminPassword": {
-      "type": "securestring",
-      "metadata": {
-        "description": "The password for the Admin Account. Must be at least 12 characters long"
-      },
-      "minLength": 12
-    },
-    "vmSize": {
-      "type": "string",
-      "metadata": {
-        "description": "Specifiy the size of VM required for the VM(s)"
-      },
-      "defaultValue": "Standard_D1_v2"
-    },
-    "vstsAccount": {
-      "type": "string",
-      "metadata": {
-        "description": "The Visual Studio Team Services account name, that is, the first part of your VSTSAccount.visualstudio.com"
-      }
-    },
-    "vstsAgentCount": {
-      "type": "int",
-      "metadata": {
-        "description": "The number of Visual Studio Team Services agents to be configured on the Virtual Machine. Default is 3"
-      },
-      "allowedValues": [
-        1,
-        2,
-        3,
-        4
-      ],
-      "defaultValue": 3
-    },
-    "vstsPersonalAccessToken": {
-      "type": "securestring",
-      "metadata": {
-        "description": "The personal access token to connect to VSTS"
-      }
-    },
-    "vstsPoolName": {
-      "type": "string",
-      "defaultValue": "Default",
-      "metadata": {
-        "description": "The Visual Studio Team Services build agent pool for this build agent to join. Use 'Default' if you don't have a separate pool."
-      }
-    },
-    "_artifactsLocation": {
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts